C/common/reading_set.cpp

```diff
@@ -192,6 +192,51 @@ ReadingSet::append(const vector<Reading *>& readings)
 	}
 }
 
+/**
+* Deep copy a set of readings to this reading set.
+*/
+bool
+ReadingSet::copy(const ReadingSet& src)
+{
+   vector<Reading *> *readings = new vector<Reading *>;
```

Comment:
This is never freed as is a memory leak, you can simply create this on the stack and then pass it to append